File: app/src/main/java/com/tifone/tnews/utils/TimeUtils.java
package com.tifone.tnews.utils;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

/**
 * Created by tongkao.chen on 2018/5/8.
 */

public class TimeUtils {
    public static String getCurrentTimeStamp() {
        return String.valueOf(System.currentTimeMillis() / 1000);
    }
    public static String getTimeStampAgo(long timeStamp) {
        SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.CHINA);
        String result = format.format(timeStamp * 1000L);
        Date date = null;
        try {
            date = format.parse(result);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return timeAgo(date);
    }
    private static String timeAgo(Date date) {
        SimpleDateFormat format = new SimpleDateFormat("MM-dd HH:mm", Locale.CHINA);
        if (date != null) {
            long agoTimeMillis = new Date(System.currentTimeMillis()).getTime() - date.getTime();
            long agoTimeMin = agoTimeMillis / 1000 / 60;
            if (agoTimeMin < 1) {
                return "刚刚";
            } else if (agoTimeMin <=60) {
                return agoTimeMin + "分钟前";
            } else if (agoTimeMin <= 60 * 24) {
                return (agoTimeMin / 60) + "小时前";
            } else if (agoTimeMin <= 60 * 24 * 2) {
                return (agoTimeMin / 60 / 24) + "天前";
            } else {
                return format.format(date);
            }
        }
        return format.format(new Date(0));
    }
}

File: app/src/main/kotlin/com.tifone.tnews/utils/TimeUtils.kt
package com.tifone.tnews.utils

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class TimeUtils {
    companion object STATIC {
        fun getCurrentTimeStamp(): String {
            return (System.currentTimeMillis() / 1000).toString()
        }

        fun getTimeStampAgo(timeStamp: Long): String {
            val format = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.CHINA)
            val result = format.format(timeStamp * 1000L)
            var date: Date? = null
            try {
                date = format.parse(result)
            } catch (e: ParseException) {
                e.printStackTrace()
            }

            return timeAgo(date)
        }

        private fun timeAgo(date: Date?): String {
            val format = SimpleDateFormat("MM-dd HH:mm", Locale.CHINA)
            if (date != null) {
                val agoTimeMillis = Date(System.currentTimeMillis()).time - date.time
                val agoTimeMin = agoTimeMillis / 1000 / 60
                return if (agoTimeMin < 1) {
                    "刚刚"
                } else if (agoTimeMin <= 60) {
                    agoTimeMin.toString() + "分钟前"
                } else if (agoTimeMin <= 60 * 24) {
                    (agoTimeMin / 60).toString() + "小时前"
                } else if (agoTimeMin <= 60 * 24 * 2) {
                    (agoTimeMin / 60 / 24).toString() + "天前"
                } else {
                    format.format(date)
                }
            }
            return format.format(Date(0))
        }
    }
}